repository_service_tuf_worker/repository.py

```diff
@@ -295,19 +317,49 @@ def _bump_version(self, role: Metadata) -> None:
         role.signed.version += 1
 
     def _bump_and_persist(
-        self, role: Metadata, role_name: str, persist: Optional[bool] = True
+        self,
+        role: Metadata,
+        role_name: str,
+        persist: Optional[bool] = True,
+        signer: Optional[Signer] = None,
+        expire: Optional[int] = None,
     ):
         """
         Bump expiry and version, sign and persist 'role' metadata into a new
         file named VERSION.rolename.json where VERSION is the new role version.
         Optionally, if persist is false, then don't persist in a file.
         """
-        self._bump_expiry(role, role_name)
+        self._bump_expiry(role, role_name, expire)
         self._bump_version(role)
-        self._sign(role)
+        self._sign(role, signer)
         if persist:
             self._persist(role, role_name)
 
+    def update_snapshot(
```

Comment:
I am not a fan that we will have `_update_snapshot` which is not used in this function called  `update_snapshot`.

Reply:
My goal is to refactor the `_update_snapshot`. I avoid doing it and right now we are adding the `update_snapshot` to avoid massive changes to it, that is already very complicated.

Reply:
Okay, we need to get this done in the future.